### PR TITLE
Fix `safeImageUrl` to force protocol only if it's missing

### PR DESCRIPTION
**Pull request by hambai**
#### Changes proposed in this Pull Request

* In #45942 I deployed a fix to `safeImageUrl` but when I think about it - we should force https only if the protocol is missing so here's a PR to fix my fix :(

#### Testing instructions

* Added a new test case for that case too - `safeImageUrl` should properly transform url with http protocol to a photon url without ssl param

### CODE REVIEW

**Comment by akirk** [comment]
Thank you for fixing this! I [already had similar handling added in my PR](https://github.com/Automattic/wp-calypso/pull/45503#discussion_r489130849) but removed it upon review request :(

**Comment by bluefuton** [comment]
Sorry @akirk, I think we got our wires crossed - I was asking for clarification of what the code did, not necessarily for it to be removed:

https://github.com/Automattic/wp-calypso/pull/45503#discussion_r489130849

Thanks for catching this @hambai.

**Comment by akirk** [comment]
Oh yes, totally my fault, didn't mean to suggest otherwise. I could not remember the exact use case so I removed it instead of digging it back up. Thanks again for resolving this!

File: client/lib/safe-image-url/index.js [hunk 1]
```diff
@@ -80,7 +80,9 @@ export default function safeImageUrl( url ) {
 		delete parsedUrl.search;
 
 		// Force https since if there's a missing protocol it'll throw an error
-		parsedUrl.protocol = 'https';
+		if ( ! parsedUrl?.protocol ) {
```
Comment (sgomes) [inline client/lib/safe-image-url/index.js:83]: No need for `?.` here; `parsedUrl` is never `null` / `undefined`. In fact, I'd rewrite all of this without an `if`, as just:

```js
parsedUrl.protocol = parsedUrl.protocol ?? 'https:';
```

**Comment by sgomes** [comment]
Was there any change here? It's been quite a while since I worked on this, and there didn't seem to be any scheme-relative URLs in use back then.

**Review by sgomes — COMMENTED** [review]
Thanks, @hambai! The changes generally look good, with a small suggestion. It looks like this has already been merged without a review, however. Presumably to fix a bug in production that's affecting users?

Speaking of which, was there any change here? It's been quite a while since I worked on this, and there didn't seem to be any scheme-relative URLs in use back then, which made it safe to apply `https` to just Automattic-owned URLs. Why is this no longer the case?

**Comment by hambai** [comment]
@sgomes ah, sorry - the context of this PR is that the uncaught error was breaking the reader on production and I tried to do an emergency fix on Saturday. Initially I forced `https` for all URLs in #45942 but decided that might create other issues so hence this PR. I wanted to merge this quickly to unbreak production so didn't put a lot of context in this particular PR.

**Comment by sgomes** [comment]
Thank you for the extra context, @hambai!

What I'm most confused about is why this started being a problem, however. There seemed to be no scheme-relative URLs in use up until recently, right? Or were these failures happening for months without anyone noticing?

**Comment by hambai** [comment]
There was a recent PR ( #45503 - Reader: Enable photon on external images with size parameters ) which changed how `safeImageUrl` works and we use that function in the reader where external image urls can come in any form so that's when the issues started.

**Comment by sgomes** [comment]
Thanks, @hambai, that makes it much clearer! 👍

**Pull request merged**